### PR TITLE
PP-13 Create new charge status USER_CANCEL_ERROR

### PR DESCRIPTION
**Pull request by mrlumbu**
## WHAT
- Creates new internal charge status USER_CANCEL_ERROR
- To be applied to any charge where user cancel has been attempted but not succeeded due to gateway error
- Maps to external charge status USER_CANCELLED, the assumption being any non terminal charges will be expired
- In case of gateway failure, error is logged
## HOW TO REVIEW
- Verify tests are ensuring correct status for all paths
## WHO
- Any dev

### CODE REVIEW

**Comment by mrlumbu** [comment]
rebuild

**Comment by rauligs** [comment]
rebuild

**Comment by mrlumbu** [comment]
rebuild

**Comment by mrlumbu** [comment]
rebuild

**Comment by mrlumbu** [comment]
rebuild

File: src/main/java/uk/gov/pay/connector/service/UserCardCancelService.java [hunk 1]
```diff
@@ -23,13 +21,14 @@ public UserCardCancelService(ChargeDao chargeDao, PaymentProviders providers, Ch
...
-        // It would be better if we had three levels of status: GatewayStatus, InternalChargeStatus and
-        // ExternalChargeStatus. This would allow us to map GatewayStatus to InternalStaus differently
-        // depending on context
-        ChargeStatus updatedStatus = (cancelResponse.getStatus().equals(ChargeStatus.SYSTEM_CANCELLED)) ?
-                getCancelledStatus() :
-                cancelResponse.getStatus();
+        ChargeStatus updatedStatus;
+        if (cancelResponse.isSuccessful()) {
+            updatedStatus = getCancelledStatus();
+
```
Comment (PauloPortugal) [inline src/main/java/uk/gov/pay/connector/service/UserCardCancelService.java]: I would remove this line

File: src/test/java/uk/gov/pay/connector/it/resources/ChargeCancelFrontendResourceITest.java [hunk 2]
```diff
@@ -85,6 +85,7 @@ public void respondWith400_whenNotCancellableState() {
                             .and()
                             .contentType(JSON)
                             .body("message", is(expectedMessage));
+
```
Comment (PauloPortugal) [inline src/test/java/uk/gov/pay/connector/it/resources/ChargeCancelFrontendResourceITest.java]: I would remove this line

**Pull request merged**